testing/python/collect.py

```diff
@@ -955,3 +955,35 @@ def check3(fix1, i, fix2):
     """)
     result = testdir.inline_run()
     result.assertoutcome(passed=6, failed=2)
+
+
+def test_dont_collect_non_function_callable(testdir):
+    modcol = testdir.getmodulecol("""
+        class Oh(object):
+            def __call__(self):
+                pass
+
+        test_a = Oh()
+
+        def test_real():
+            pass
+    """)
+    colitems = modcol.collect()
+    assert len(colitems) == 1
+
+
+def test_dont_run_non_function_callable(testdir):
```

Comment:
Oh, also add the reference to the issue in the docstring, along with a short description if possible. :smile: